src/server/get_scale.rs

```diff
@@ -0,0 +1,149 @@
+                },
+                _ => {
+                },
+            }
+        }
+    }
+}
+
+impl Dispatch<wl_output::WlOutput, ()> for AppData {
+    fn event(
+        data: &mut Self,
+        _: &wl_output::WlOutput,
+        event: wl_output::Event,
+        _: &(),
+        _: &Connection,
+        _: &QueueHandle<AppData>,
+    ) {
+        if let wl_output::Event::Mode { flags, width, height, refresh } = event {
+            data.physical_width = width;
+            data.physical_height = height;
+        }
+    }
+}
+
+impl Dispatch<ZxdgOutputManager, ()> for AppData {
+    fn event(
+        _: &mut Self,
+        _: &ZxdgOutputManager,
+        _: ZxdgOutputManagerEvent,
+        _: &(),
+        _: &Connection,
+        _: &QueueHandle<AppData>,
+    ) {
+    }
+}
+
+impl Dispatch<ZxdgOutput, ()> for AppData {
+    fn event(
+        data: &mut Self,
+        _: &ZxdgOutput,
+        event: ZxdgOutputEvent,
+        _: &(),
+        _: &Connection,
+        _: &QueueHandle<AppData>,
+    ) {
+        if let ZxdgOutputEvent::LogicalSize { width, height } = event {
+            data.logical_width = width;
+            data.logical_height = height;
+        }
+    }
+}
+
+// The main function of our program
+pub fn get_scale() -> f64 {
+    let mut scale = 1.0;
+    // Create a Wayland connection by connecting to the server through the
+    // environment-provided configuration.
+    let conn = Connection::connect_to_env();
+    if let Ok(conn) = conn {
+        // Retrieve the WlDisplay Wayland object from the connection. This object is
+        // the starting point of any Wayland program, from which all other objects will
+        // be created.
+        let display = conn.display();
+
+        // Create an event queue for our event processing
+        let mut event_queue = conn.new_event_queue();
+        // And get its handle to associated new objects to it
+        let qh = event_queue.handle();
+
+        // Create a wl_registry object by sending the wl_display.get_registry request
+        // This method takes two arguments: a handle to the queue the newly created
+        // wl_registry will be assigned to, and the user-data that should be associated
+        // with this registry (here it is () as we don't need user-data).
+        let _registry = display.get_registry(&qh, ());
+
+        let mut data = AppData::default();
+
+        let ret = event_queue.blocking_dispatch(&mut data);
+        if ret.is_err() {
+            return scale;
+        }
+
+        if let (Some(wl_output), Some(xdg_output_manager)) = (&data.wl_output, &data.xdg_output_manager) {
+            xdg_output_manager.get_xdg_output(&wl_output, &qh, ());
+        }
+
+        let ret = event_queue.blocking_dispatch(&mut data);
+        if ret.is_err() {
+            return scale;
+        }
+
+        scale = data.physical_width as f64 / data.logical_width as f64;
+    }
+
+    scale
```

Comment:
I think you should return 0 if it fails. Then use old value if this is 0.

Reply:
> Ubuntu 24.04 Gnome ❌ Broken Current version works fine
Will it return 0 in this case?

Reply:
> > Ubuntu 24.04 Gnome ❌ Broken Current version works fine
> 
> Will it return 0 in this case?

No. Because the cursor's positions devide 1.5 in the video.

Reply:
> I think you should return 0 if it fails. Then use old value if this is 0.

1.0 means no scale, so the rdp mouse cursor will move distance which it received.

Reply:
What scale your code return?

Reply:
> Will it return 0 in this case?

If you do not like to return 0 as failure to get scale, you can return Err.

Reply:
There is no difference between `1.0` or err returned, return err we will disable mouse cursor scale, return 1.0 it's same too.

Reply:
Yes, now it will return error when can't get this scale.